cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/SystemDatasetRuntimeModule.java

```diff
@@ -88,7 +121,7 @@ protected void configure() {
                                                                             Names.named("defaultDatasetModules"));
 
         // NOTE: order is important due to dependencies between modules
-        mapBinder.addBinding("orderedTable-hbase").toInstance(new HBaseTableModule());
+        mapBinder.addBinding("orderedTable-hbase").toInstance(tableModule);
```

Comment:
It is necessary for the `CConfiguration` to be injected because the `ConfigModule` can do modification to the `CConfiguration`. Also, by using provider, you don't need to change existing places that uses `StreamAdminModules` / `SystemDatasetRuntimeModule`.